src/src/com/microsoft/aad/adal/CacheKey.java

```diff
@@ -106,54 +125,55 @@ public static String createCacheKey(String authority, String resource, String cl
     }
 
     /**
-     * @param item Token item in the cache
-     * @return CacheKey to save token
+     * Create cachekey from {@link TokenCacheItem}. It will use {@link UserInfo#getUserId()} 
+     * as the user for cachekey if present. 
+     * @param item {@link TokenCacheItem} that is used to create the cache key. 
+     * @return String value of the {@link CacheKey} to save token. 
+     * @throws AuthenticationException 
      */
-    public static String createCacheKey(TokenCacheItem item) {
+    public static String createCacheKey(TokenCacheItem item) throws AuthenticationException {
         if (item == null) {
             throw new IllegalArgumentException("TokenCacheItem");
         }
 
         String userid = null;
-
         if (item.getUserInfo() != null) {
             userid = item.getUserInfo().getUserId();
         }
-
-        return createCacheKey(item.getAuthority(), item.getResource(), item.getClientId(),
-                item.getIsMultiResourceRefreshToken(), userid);
+        
+        final TokenEntryType tokenEntryType = item.getTokenEntryType();
+        switch (tokenEntryType) {
+        case REGULAR_TOKEN_ENTRY: 
+            return createCacheKeyForRTEntry(item.getAuthority(), item.getResource(), 
+                    item.getClientId(), userid);
+        case MRRT_TOKEN_ENTRY:
+            return createCacheKeyForMRRT(item.getAuthority(), item.getClientId(), userid);
+        case FRT_TOKEN_ENTRY:
+            return createCacheKeyForFRT(item.getAuthority(), item.getFamilyClientId(), userid);
+        default: 
+            throw new AuthenticationException(ADALError.INVALID_TOKEN_CACHE_ITEM, "Cannot create cachekey from given token item");
+        }
     }
-
+    
     /**
-     * @param item AuthenticationRequest item
-     * @param cacheUserId UserId in the cache
-     * @return CacheKey to save token
+     * Create cache key for regular RT entry. 
      */
-    public static String createCacheKey(AuthenticationRequest item, String cacheUserId) {
-        if (item == null) {
-            throw new IllegalArgumentException("AuthenticationRequest");
-        }
-
-        return createCacheKey(item.getAuthority(), item.getResource(), item.getClientId(), false,
-                cacheUserId);
+    public static String createCacheKeyForRTEntry(final String authority, final String resource, final String clientId, final String userId) {
+        return createCacheKey(authority, resource, clientId, false, userId, null);
     }
```

Comment:
nope, I'll then need to take in some unnecessary params in the single method entry. 

---
In reply to: [63265550](https://github.com/AzureAD/azure-activedirectory-library-for-android/pull/611#discussion_r63265550) [](ancestors = 63265550)